regcomp.h

```diff
@@ -150,7 +186,12 @@ struct regnode_p {
     U8	flags;
     U8  type;
     U16 next_off;
-    SV * arg1;
+#if PTRSIZE == 8
+    U32 ptr_high;
+    U32 ptr_low;
+#else
```

Comment:
Instead of this, why not just `char ptr_bytes[sizeof(SV*)]` and memcpy() the pointer value in and out?

ARGp() would need to be (or use) an inline function or gcc brace group, but I think it would simplify things otherwise, and make fewer assumptions about pointer sizes.

The compiler can also optimize this to a simple (unaligned) access, eg.

```
#include <string.h>

typedef unsigned char U8;
typedef unsigned short U16;

struct regnode_p {
    U8	flags;
    U8  type;
    U16 next_off;
    char arg1_bytes[sizeof(void *)];
};

static inline void
ARGp_SET(struct regnode_p *n, void *p) {
    memcpy(n->arg1_bytes, &p, sizeof(p));
}

static inline void *
ARGp(const struct regnode_p *n) {
    void *p;
    memcpy(&p, n->arg1_bytes, sizeof(p));

    return p;
}

int x;

void foo(struct regnode_p *n) {
    ARGp_SET(n, &x);
}

void *bar(struct regnode_p *n) {
    return ARGp(n);
}
```
With gcc -O2, foo and bar are:
```
foo:
.LFB2:
        .cfi_startproc
        leaq    x(%rip), %rax
        movq    %rax, 4(%rdi)
        ret
        .cfi_endproc
...
bar:
.LFB3:
        .cfi_startproc
        movq    4(%rdi), %rax
        ret
        .cfi_endproc
```

As you can see, while it's undefined behaviour for us to access the unaligned data in C, the compiler has more freedom.

Reply:
I dont really understand your argument. Why would using memcpy to copy 8 bytes be better than assigning two 32 bit words? 

Ideally we Just Shouldn't Have 64 bit members in our regnodes. Id rather we just eliminate the need than make our implementation better, this is just a work around for now until I have the time to address this properly and rip it out entirely. If had been paying attention when this was introduced I would have blocked it and directed Karl to the data[] array and related functionality.

Reply:
I was insufficiently caffeinated when I wrote my original reply.  Roger, willco. :-)